pkg/agent/agent.go

```diff
@@ -198,6 +201,25 @@ func InitializeWithAgentFeatures(centralCfg config.CentralConfig, agentFeaturesC
 	return nil
 }
 
+func registerExternalIDPs() {
+	if agent.cfg.GetAgentType() != config.TraceabilityAgent {
+		idPCfg := agent.agentFeaturesCfg.GetExternalIDPConfig()
+		tlsCfg := agent.cfg.GetTLSConfig()
+		proxy := agent.cfg.GetProxyURL()
+		timeout := agent.cfg.GetClientTimeout()
+		for _, idp := range idPCfg.GetIDPList() {
+			registerCredentialProvider(idp, tlsCfg, proxy, timeout)
+		}
+	}
+}
+
+func registerCredentialProvider(idp config.IDPConfig, tlsCfg config.TLSConfig, proxyURL string, clientTimeout time.Duration) {
+	err := GetAuthProviderRegistry().RegisterProvider(idp, tlsCfg, proxyURL, clientTimeout)
+	if err != nil {
+		log.Errorf("unable to register external IdP provider, any credential request to the IdP will not be processed. %s", err.Error())
```

Comment:
Can you please use the logger in this file

Reply:
Will do the change